sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DistinctDocumentQueryExecutionContext.java

```diff
@@ -20,7 +19,9 @@
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.BiFunction;
 
-public class DistinctDocumentQueryExecutionContext<T extends Resource> implements IDocumentQueryExecutionComponent<T> {
+public class DistinctDocumentQueryExecutionContext<T>
+    implements IDocumentQueryExecutionComponent<T> {
```

Comment:
is the rule is that:
if there is any special properties need from document, then we are using Document type, otherwise, we are going to use T? 

But sometimes, we are having executionContext on top of another executionContext. for example, GroupByContext can use DistinctQueryExecutionContext as the base component, if we already transformed to T, then how it is going to use the Document type in GroupByExecutionContext?

Reply:
sorry ignore me, I think I just need to read this part logic a little bit more